pkg/domain/infra/tunnel/runtime.go

```diff
@@ -18,3 +25,45 @@ type ContainerEngine struct {
 type SystemEngine struct {
 	ClientCtx context.Context
 }
+
+func ProxySignal(ctrID string, killFunc func(string) error) {
+	sigBuffer := make(chan os.Signal, terminal.SignalBufferSize)
+	signal.CatchAll(sigBuffer)
+
+	logrus.Debugf("Enabling signal proxying")
+
+	go func() {
+		for s := range sigBuffer {
+			// Ignore SIGCHLD and SIGPIPE - these are mostly likely
+			// intended for the podman command itself.
+			// SIGURG was added because of golang 1.14 and its preemptive changes
+			// causing more signals to "show up".
+			// https://github.com/containers/podman/issues/5483
+			if s == syscall.SIGCHLD || s == syscall.SIGPIPE || s == syscall.SIGURG {
+				continue
+			}
+
+			signalName, err := signal.ParseSysSignalToName(s.(syscall.Signal))
+			if err != nil {
+				logrus.Infof("Ceasing signal forwarding to container %s as it has stopped", ctrID)
+			}
+			if err := killFunc(signalName); err != nil {
+				if err.Error() == define.ErrCtrStateInvalid.Error() {
+					logrus.Infof("Ceasing signal forwarding to container %s as it has stopped", ctrID)
+				} else {
+					logrus.Errorf("forwarding signal %d to container %s: %v", s, ctrID, err)
+				}
+				// If the container dies, and we find out here,
+				// we need to forward that one signal to
+				// ourselves so that it is not lost, and then
+				// we terminate the proxy and let the defaults
+				// play out.
+				signal.StopCatch(sigBuffer)
+				if err := syscall.Kill(syscall.Getpid(), s.(syscall.Signal)); err != nil {
```

Comment:
Doesn't this have to happen on the server side?

Reply:
I don't think so. @mheon am I right?